### PR TITLE
Implement deleteTrain

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This is part of the overall game command for vehicle sell and will replace the vehicleHead::removeDanglingTrain stub function.

### CODE REVIEW

File: src/OpenLoco/Vehicles/VehicleManager.cpp [hunk 1]
```diff
@@ -14,6 +16,37 @@ namespace OpenLoco::VehicleManager
...
+            Ui::Windows::VehicleList::removeTrainFromList(vehListWnd, head.id); // 0x004C1D19
+        }
+        // Change to vanilla, update the build window to a valid train
+        auto* vehBuildWnd = Ui::WindowManager::find(Ui::WindowType::buildVehicle, enumValue(head.owner));
+        if (vehBuildWnd != nullptr)
+        {
+            vehBuildWnd->invalidate();
+            Ui::Windows::BuildVehicle::sub_4B92A5(vehBuildWnd);
+        }
+
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/VehicleManager.cpp:37]: Add a part to remove audio refs as well.

Comment (LeftofZen) [reply src/OpenLoco/Vehicles/VehicleManager.cpp:67]: Was this added? I don't see it

File: src/OpenLoco/Vehicles/VehicleManager.cpp [hunk 2]
```diff
@@ -124,3 +209,31 @@ namespace OpenLoco::Vehicles::RoutingManager
...
+            if (head->orderTableOffset >= removeOrderTableOffset)
+            {
+                head->orderTableOffset += sizeOfRemovedOrderTable;
+            }
+        }
+    }
+
+    // 0x00470334
+    // Remove vehicle ?orders?
+    void sub_470334(VehicleHead* const head)
```
Comment (LeftofZen) [inline src/OpenLoco/Vehicles/VehicleManager.cpp:231]: can we rename this method and remove the comment? I think it's clear it's clearing orders?

**Review by LeftofZen — APPROVED** [review]
Playtested, no issues with deleting trains/trams/trucks/planes, or individual train vehicles, or crashed trains

File: src/OpenLoco/Vehicles/VehicleManager.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Vehicles/VehicleManager.cpp:67]: No not yet :(

**Pull request merged**